src/Listener.php

```diff
@@ -64,10 +64,48 @@ public function setTagNameCase($case)
      *                                 It uses array_walk() internally.
      *                                 The suggested signature for the argument is:
      *                                 function (string &$value, string $tag);
+     *
+     * @deprecated
      */
     public function setTagValueProcessor(callable $processor = null)
     {
-        $this->tagValueProcessor = $processor;
+        if (is_null($processor)) {
+            $this->tagValueProcessors = [];
+
+            return;
+        } // else
+        $this->tagValueProcessors = [$processor];
+    }
+
+    /**
+     * @param  $processor Function or array of functions to be applied to every member
+     *                    of an BibTeX entry. Uses array_walk() internally.
+     *                    The suggested signature for each function argument is:
+     *                        function (string &$value, string $tag);
+     *                    Note that functions will be applied in the same order
+     *                    in which they were added.
+     * @throws \InvalidArgumentException
+     */
+    public function addTagValueProcessor($processor)
+    {
+        if (is_array($processor)) {
```

Comment:
The problem of accepting multiple functions at the same time is that an array may be also a callable, see the PHP docs http://php.net/manual/en/language.types.callable.php#example-75  types 2 and 3